fastlane_bot/helpers/txhelpers.py

```diff
@@ -969,13 +972,57 @@ def approve_token_for_arb_contract(self, token_address: str, approval_amount: in
                             base_gas_price=baseFee, max_priority_fee=max_priority, nonce=self.get_nonce()
                         )
                     )
+                    approve_tx = self.sign_transaction(approve_tx)
+                    self.ConfigObj.logger.info(f"Submitting approval for token: {token_address}")
+                    return self.submit_transaction(approve_tx)
                 except Exception as e:
                     self.ConfigObj.logger.info(
                         f"(***2***) Error when building transaction: {e.__class__.__name__} {e}")
             else:
                 return None
-        self.ConfigObj.logger.info(f"Submitting approval for token: {token_address}")
-        return self.submit_transaction(approve_tx)
 
+    async def get_layer_one_gas_fee(self, signed_transaction) -> Decimal:
+        """
+        Returns the expected layer one gas fee for a layer 2 Optimism transaction
+        :param signed_transaction: the signed ethereum TX
+
+        returns: Decimal
+            The total fee (in gas token) for the l1 gas fee
+        """
+        ethereum_base_fee = await self._get_layer_one_gas_price()
+        fixed_overhead = await self._get_layer_one_fee_overhead()
+        dynamic_overhead = await self._get_layer_one_fee_scalar()
+        zero_bytes, non_zero_bytes = count_bytes(signed_transaction["rawTransaction"])
+        tx_data_gas = zero_bytes * 4 + non_zero_bytes * 16
+        tx_total_gas = (tx_data_gas + fixed_overhead) * dynamic_overhead
+        l1_data_fee = tx_total_gas * ethereum_base_fee
+        l1_data_fee = Decimal(str(l1_data_fee)) / Decimal("1000000") / Decimal(str(10 ** 18))
```

Comment:
I'm pretty sure that 'fee' is always denoted in wei units, which means that you should not be dividing by `10 ** 18`.
I also have no idea why you chose to divide by 1000000, which sounds related to the fee resolution on some DEXs (Carbon and Bancor, for example), but has absolutely nothing to do with gas fee in any way.
In short, it seems possible that both of these two divisions are not in place.
But if they were, then you may as well just divide by their multiplication, i.e., by `10 ** 24`.

Reply:
Dividing by 10 ** 18:
The gas cost we calculated before this is in decimal format in order to compare it to the estimated profit in gas tokens. To add the l1 gas fee to it, it needs to be in the same format. 

Dividing by 1,000,000:
The l1FeeScalar returns a value in PPM, for example: 684000. So we need to multiply by 0.684. I chose to add the division at the end because everything until that last line is in int format. 

I'm changing this to divide by 10 ** 24

Reply:
> for example: 684000. So we need to multiply by 0.684

Sorry, but this example makes zero sense.

The value of `684000 * 0.684` is `467856`. How exactly is that related to PPM resolution?